src/Common/src/System/Data/Common/FieldNameLookup.cs

```diff
@@ -142,7 +138,7 @@ private int LinearIndexOf(string fieldName, CompareOptions compareOptions)
         private void GenerateLookup()
         {
             int length = _fieldNames.Length;
-            Hashtable hash = new Hashtable(length);
+            Dictionary<string, int> hash = new Dictionary<string, int>(length);
```

Comment:
Nit: There is a comment above the `GenerateLookup` method that refers to "Hashtable" which could be changed to "Dictionary".